### ISSUE TITLE
Compatibility with segment replication

### ISSUE THREAD

**Issue by dreamer-89** [opensearch-project/security-dashboards-plugin#1494]
### Summary
With 2.9.0 release, there are lot of enhancements going in for segment replication[1][2] feature (went GA in 2.7.0), we need to ensure different plugins are compatible with current state of this feature. Previously, we ran [tests](https://github.com/opensearch-project/OpenSearch/issues/6602) on plugin repos to verify this compatibility but want plugin owners to be aware of these changes so that required updates (if any) can be made

#### 1. Refresh policy behavior
1. RefreshPolicy.IMMEDIATE will only refresh primary shards but not replica shards immediately. Instead post refresh, primary will start  a round of segment replication to update the replica shard copies leading to eventual consistency.
2. RefreshPolicy.WAIT_UNTIL ensures the indexing operation is searchable in your cluster i.e. RAW (Read after write guarantee). With segment replication, this guarantee is not promised due to delay in replica shared updates from asynchronous background refreshes.

####  2. Refresh lag on replicas
With segment replication, there is inherent delay in documents to be searchable on replica shard copies. This is due to the fact that replica shard copies over data (segment) files from primary. Thus, compared to document replication, there will be on average increase in amount of time the replica shards are consistent with primaries.

#### 3. System/hidden indices support
With https://github.com/opensearch-project/OpenSearch/pull/8200, system and hidden indices are now supported with `SEGMENT` replication strategy. We need to ensure there are no bottlenecks which prevents system/hidden indices with segment replication.

### Next steps
With segment replication strong reads are not guaranteed. Thus, if the plugin needs strong reads guarantees specially as alternative to change in behavior of refresh policy and lag on replicas (point 1 and 2 above), we need to update search requests to target primary shard only. With https://github.com/opensearch-project/OpenSearch/pull/7375, core now supports primary shards only based search. Please follow [documentation](https://opensearch.org/docs/latest/api-reference/search/#the-preference-query-parameter) for examples and details

### Open questions
In case of any questions or issues, please post it in [core](https://github.com/opensearch-project/OpenSearch/issues/8211) issue

### Reference
[1] [Design](https://github.com/opensearch-project/OpenSearch/issues/2229)
[2] [Documentation](https://opensearch.org/docs/latest/tuning-your-cluster/availability-and-recovery/segment-replication/index/)

**Comment by dreamer-89** [comment]
Request owners to add `v2.9.0` label on this issue.

**Comment by scrawfor99** [comment]
[Triage] This has been addressed @RyanL1997 confirmed that the existing behavior is maintained as part of the previous test and @cwperks addresses this with: https://github.com/opensearch-project/security/pull/2903.

**Comment by dreamer-89** [comment]
> Closing this issue for the security-dashboards-plugin repo. Any work to accommodate segrep in the security plugin would be done on the backend repo.

@cwperks : By backend repo, did you mean security repository ? I see previously [issue](https://github.com/opensearch-project/security/issues/2916) was closed in lieu of different issue. 

security issue: https://github.com/opensearch-project/security/issues/2916

**Comment by cwperks** [comment]
@dreamer-89 Yes, the security repo. The security-dashboards-plugin repo utilizes the `.kibana-*` indices to support multi-tenancy, but those indices are not system indices the way that `.opendistro_security` (the security index) is. The `security` repo requires strong consistency for the security index to ensure that every node reading from the security index sees the freshest data from the index. There was a PR opened in the security repo to set the preference of the mget request that the security plugin performs when reading from the security index, but the PR was closed. See discussion on the PR here: https://github.com/opensearch-project/security/pull/2903

security issue: https://github.com/opensearch-project/security/issues/2916

**Comment by dreamer-89** [comment]
Moving discussion to security issue https://github.com/opensearch-project/security/issues/2916
